utensor_cgen/backend/utensor/_backend_impl.py

```diff
@@ -1,39 +1,45 @@
 from utensor_cgen.backend.base import Backend
-from utensor_cgen.utils import (
-  class_property, parse_toml,
-  LazyLoader, LazyAttrib,
-  Configuration,
-)
+from utensor_cgen.utils import (LazyAttrib, LazyLoader, class_property,
+                                parse_toml)
 
 code_generator = LazyLoader(submod_name='backend.utensor.code_generator')
-_graph_lower = LazyLoader(submod_name='backend.utensor._graph_lower')
+transformer = LazyLoader(submod_name='backend.transformer')
+_op_lower = LazyLoader(submod_name='backend.utensor._graph_lower._op_lower')
+generic_graph_lower = LazyLoader(submod_name='backend.graph_lower')
 uTensorLegacyCodeGenerator = LazyAttrib(code_generator, 'uTensorLegacyCodeGenerator')
 uTensorRearchCodeGenerator = LazyAttrib(code_generator, 'uTensorRearchCodeGenerator')
-uTensorLegacyGraphLower = LazyAttrib(_graph_lower, 'uTensorLegacyGraphLower')
-uTensorRearchGraphLower = LazyAttrib(_graph_lower, 'uTensorRearchGraphLower')
+uTensorLegacyGraphLower = LazyAttrib(_op_lower, 'uTensorLegacyGraphLower')
+uTensorRearchGraphLower = LazyAttrib(_op_lower, 'uTensorRearchGraphLower')
+TensorAllocationPlanner = LazyAttrib(generic_graph_lower, 'TensorAllocationPlanner')
+PipelineTransformer = LazyAttrib(transformer, 'PipelineTransformer')
 
-del code_generator, _graph_lower
+del code_generator, _op_lower, generic_graph_lower
 
 class uTensorBackend(Backend):
```

Comment:
This class is SUPER important as an example backend, but it is missing some key docs like how it links the middleware bits to the target backend.

Reply:
Right, I'm about to update the architecture of the cli in the docs.
By the time the backend is written, I'm trying to come up with a proper way to do it so I didn't write any docs.